File: bench/programs/sieve/sieve.go
package main

import (
	"fmt"
	"time"
)

func run() int64 {
	n := 2000000
	sieve := make([]bool, n)
	for i := range sieve {
		sieve[i] = true
	}
	var count int64 = 0
	for p := 2; p < n; p++ {
		if sieve[p] {
			count++
			for m := p * 2; m < n; m += p {
				sieve[m] = false
			}
		}
	}
	return count
}

func main() {
	run() // warmup
	t0 := time.Now()
	ans := run()
	dt := time.Since(t0).Nanoseconds()
	fmt.Println(ans)
	fmt.Println(dt)
}

File: bench/programs/sieve/sieve.py
import time


def run():
    n = 2000000
    sieve = bytearray([1]) * n
    count = 0
    for p in range(2, n):
        if sieve[p]:
            count += 1
            for m in range(p * 2, n, p):
                sieve[m] = 0
    return count


run()  # warmup
t0 = time.perf_counter_ns()
ans = run()
dt = time.perf_counter_ns() - t0
print(ans)
print(dt)
